### PR TITLE
[Performance] Always render a consistent number of elements

### PR DESCRIPTION
**Pull request by g-cassie**
TLDR: this modification improves performance when an ember-collection is scrolled for the first time after being rendered.

Forgive my loose terminology below - let me know if I am not making sense.

Assume a browser displays a vertically scrolling `ember-collection` that has 20 elements on screen and has a buffer value set to 5.  Currently, when this collection first renders, it will "initial-render" 20 elements on screen and an additional 5 elements off the bottom of the screen.  As soon as the user scrolls 5 elements down, it will have to "initial-render" an additional 5 elements off the top of the screen.  The other 25 elements that are already render will be "recycle-rendered" which is much faster.  The rendering of these 5 new elements can result in noticeable lag when the user first scrolls in a collection consisting of complex applications.  

In my use case, "initial render" of an element takes about ~20ms whereas "recycle-render" takes about ~3ms.  That 5 x ~20ms when scrolling results in choppy scrolling.  Scrolling is much smoother with this change in place.  I'm not really sure how I can quantify this more empirically.

I am submitting this for feedback before I do the work to fix the broken unit tests.  If there is support I will go ahead and get it ready for merge.  It may also make sense to implement the corresponding behaviour when starting at the end of the collection - though obviously this is more of an edge case.

### CODE REVIEW

**Comment by g-cassie** [comment]
After looking at this further I think you could get even better performance by also ensuring that `ember-collection` never renders less cells than it did in its previous render cycle.  This is important in any list that has variable heights/widths such that you may have 20 elements on screen or 40.  In this case you will get much smoother scroll if you always leave 40 cells rendered as there will be no "initial renders" after the `ember-collection` renders for the first time.

**Comment by ewwilson** [comment]
This appears to have corrected some issues that I was experiencing so it would be nice to get this change in an official release.
